File: Food_Delivery_Microservices/api/src/main/java/io/github/marbys/api/composite/RestaurantAggregate.java
package io.github.marbys.api.composite;

import java.util.List;

public class RestaurantAggregate {
    private final int restaurantId;
    private final String name;
    private final int rating;
    private final List<DishSummary> menu;
    private final List<OrderSummary> orders;
    private final ServiceAddresses serviceAddresses;

    public RestaurantAggregate() {
        this.restaurantId = 0;
        this.name = null;
        this.rating = 0;
        this.menu = null;
        this.orders = null;
        this.serviceAddresses = null;
    }
    public RestaurantAggregate(int restaurantId, String name, int rating, List<DishSummary> menu, List<OrderSummary> orders, ServiceAddresses serviceAddresses) {
        this.restaurantId = restaurantId;
        this.name = name;
        this.rating = rating;
        this.menu = menu;
        this.orders = orders;
        this.serviceAddresses = serviceAddresses;
    }

    public int getRestaurantId() {
        return restaurantId;
    }

    public String getName() {
        return name;
    }

    public int getRating() {
        return rating;
    }

    public List<DishSummary> getMenu() {
        return menu;
    }

    public List<OrderSummary> getOrders() {
        return orders;
    }
    public ServiceAddresses getServiceAddresses() {
        return serviceAddresses;
    }
}

File: food-delivery-microservices-frontend/src/app/restaurant-aggregate.ts
import {ServiceAddresses} from "./service-addresses";
import {Order} from "./order";
import {Menu} from "./menu";

  export class RestaurantAggregate {
    menu: Menu[];
    name: string;
    orders: Order[];
    rating: number;
    restaurantId: number;
    serviceAddresses: ServiceAddresses;


    constructor(menu: Menu[], name: string, orders: Order[], rating: number, restaurantId: number, serviceAddresses: ServiceAddresses) {
      this.menu = menu;
      this.name = name;
      this.orders = orders;
      this.rating = rating;
      this.restaurantId = restaurantId;
      this.serviceAddresses = serviceAddresses;
    }
  }

